tests/at_functional_test/test/atclient_delete_test.dart

```diff
@@ -0,0 +1,69 @@
+import 'dart:io';
+import 'dart:typed_data';
+
+import 'package:at_client/at_client.dart';
+import 'package:at_commons/at_commons.dart';
+import 'package:test/test.dart';
+
+import 'set_encryption_keys.dart';
+import 'test_utils.dart';
+
+/// The tests verify the delete key functionality of at_client
+void main() {
+  test('delete method - public key', () async {
+    var atsign = '@alice🛠';
+    var preference = TestUtils.getPreference(atsign);
+    final atClientManager = await AtClientManager.getInstance()
+        .setCurrentAtSign(atsign, 'wavi', preference);
+    var atClient = atClientManager.atClient;
+    atClientManager.syncService.sync();
+    // To setup encryption keys
+    await setEncryptionKeys(atsign, preference);
+    // public:phone.wavi@alice🛠
+    var phoneKey = AtKey()
+      ..key = 'location'
+      ..metadata = (Metadata()..isPublic = true)
+      ..sharedBy = atsign;
+    var value = 'California';
+    var putResult = await atClient.put(phoneKey, value);
+    expect(putResult, true);
+    var getResult = await atClient.delete(phoneKey);
+    expect(getResult, true);
+  });
+  // uncomment this test once server is released with PR #966 changes
+  // test('delete method - delete non existent key', () async {
+  //   var atsign = '@alice🛠';
+  //   var preference = TestUtils.getPreference(atsign);
+  //   final atClientManager = await AtClientManager.getInstance()
+  //       .setCurrentAtSign(atsign, 'wavi', preference);
+  //   var atClient = atClientManager.atClient;
+  //   atClientManager.syncService.sync();
+  //   // To setup encryption keys
+  //   await setEncryptionKeys(atsign, preference);
+  //   // phone.me@alice🛠
+  //   var phoneKey = AtKey()
+  //     ..key = 'location'
+  //     ..metadata = (Metadata()..isPublic = true)..sharedBy=atsign;
+  //   var value = 'California';
+  //   var putResult = await atClient.put(phoneKey, value);
+  //   expect(putResult, true);
+  //   var getResult = await atClient.delete(phoneKey);
```

Comment:
1. A naming thing - should name this variable 'deleteResult' instead of 'getResult'
2. You can check server version and assert greater than x.y.z instead of commenting out the test. See here for existing examples of usage
```
find . -name "*.dart" -exec egrep -l 'getVersion *\(\)' {} \;                 
./at_server/tests/at_end2end_test/test/update_verb_test.dart
./at_server/tests/at_end2end_test/test/e2e_test_utils.dart
./at_server/tests/at_end2end_test/test/plookup_verb_test.dart
./at_server/tests/at_end2end_test/test/notify_verb_test.dart
```

Reply:
i have implemented the test. Ran into this issue while running the tests
 https://github.com/atsign-foundation/at_server/issues/1005
 Had a discussion with @kalluriramkumar and tried  a fix. Have to verify the fix with Chris/Xavier.

Reply:
issue is fixed